test/e2e/framework/network_pods.go

```diff
@@ -81,16 +81,16 @@ func (np *NetworkPod) AwaitReady() {
 
 	np.Pod = AwaitUntil("get pod", func() (interface{}, error) {
 		return pods.Get(np.Pod.Name, metav1.GetOptions{})
-	}, func(result interface{}) (bool, error) {
+	}, func(result interface{}) (bool, string, error) {
 		pod := result.(*v1.Pod)
 		if pod.Status.Phase != v1.PodRunning {
 			if pod.Status.Phase != v1.PodPending {
-				return false, fmt.Errorf("expected pod to be in phase \"Pending\" or \"Running\"")
+				return false, "", fmt.Errorf("expected pod to be in phase \"Pending\" or \"Running\"")
 			}
-			return false, nil // pod is still pending
+			return false, "Pod is still pending", nil
 		}
 
-		return true, nil // pod is running
+		return true, "", nil // pod is running
```

Comment:
Maybe also return "pod is running"

Reply:
The message is intended as additional error/status context thus is only used if `false' or en error is returned.